watertap/tools/recursive_parameter_sweep.py

```diff
@@ -0,0 +1,293 @@
+###############################################################################
+# WaterTAP Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+import numpy as np
+import pyomo.environ as pyo
+import sys
+import os
+import itertools
+import warnings
+import copy, pprint
+import h5py
+
+from scipy.interpolate import griddata
+from enum import Enum, auto
+from abc import abstractmethod, ABC
+from idaes.core.util import get_solver
+
+from idaes.surrogate.pysmo import sampling
+from idaes.core.util.model_statistics import (
+    variables_in_activated_equalities_set,
+    expressions_set,
+    total_objectives_set,
+)
+from pyomo.core.base.block import TraversalStrategy
+from watertap.tools.parameter_sweep import (
+    _aggregate_results_arr,
+    _build_combinations,
+    _create_local_output_skeleton,
+    _create_global_output,
+    _default_optimize,
+    _divide_combinations,
+    _do_param_sweep,
+    _init_mpi,
+    _process_sweep_params,
+    _process_results_filename,
+    _save_results,
+    _write_outputs,
+)
+
+np.set_printoptions(linewidth=200)
```

Comment:
I know we do this in the other file ... what's the reasoning? Generally we importing a module shouldn't have weird side effects like this.

If it's needed, perhaps we can write a little context manager to modify this value for when its needed and reset it automatically.

Reply:
Setting a wider line width for numpy print options simply enables easier debugging of code using print statements as it doesn't wrap numpy arrays. The 200 character width is definitely arbitrary in the sense that it was sufficient in making sure the output was easily readable for debugging and understanding the code. Since we don't really print numpy arrays during analysis runs, it doesn't affect the outputs.

I am open to having a mechanism to change the line width value. That said I just disagree with the default value of 75 characters in this day and age of HD and 4K screens.

Reply:
If the answer is a matter of debugging print statements, I would suggest we eliminate this line in this file and `parameter_sweep.py`.

It's easy enough to change when we're working on the code an removing it once done, or write a one-off function for printing which uses the context manager instead: https://numpy.org/doc/stable/reference/generated/numpy.printoptions.html#numpy.printoptions. We should not modify the options in a widely-used third-party package globally without a very compelling reason.

Reply:
Removed it in the most recent commit.